Explain editing of additional app charts

andreas-kupries opened the pull request:
Fix [#1271](https://github.com/epinio/epinio/issues/1271).

Commit:
[WIP] Document how to edit / create additional app charts

andreas-kupries committed:
Rebased. Moved new file to new location for pages.

andreas-kupries committed:
Rework the text ...

- notes, questions, todos to the top to be easily seen while working on the page
- rephrased the intro a bit
- removed operator/dev tags from sections title, and moved under proper separate sections per role
- filled the dev sections

andreas-kupries committed:
removed leftover cli reference page, this is all under docs/ now.

andreas-kupries committed:
completed docs, split per role, moved to how-tos.

nunix commented on docs/howtos/create_custom_appcharts.md:
```diff
@@ -0,0 +1,150 @@
+The tarball is placed into the current working directory and has the form
+`application-VERSION.tar.gz` where `VERSION` is the chart version.
+
+Note that in this How-To we have not attempted to change the chart version. Versioned
+chart development is out of scope here.
```
```suggestion
> Note: we didn't change the chart version. Versioned
chart development is out of scope for this How-To.
```

nunix commented on docs/howtos/create_custom_appcharts.md:
```diff
@@ -0,0 +1,150 @@
+The first step in making the new chart created in the previous section available to Epinio
+is to place the generated tarball on some publicly accessible server.
+
+Given the plethora of possible options this How-To simply assumes that the tarball is
+available at the example URL
```
```suggestion
Given the plethora of possible options, this How-To simply assumes that the tarball is
available at the example URL:
```

nunix commented on docs/howtos/create_custom_appcharts.md:
````diff
@@ -0,0 +1,150 @@
+https://mydomain.org/epinio-application-0.1.15-fluentd.tgz
+```
+
+Now save
````
```suggestion
Copy the following text, change the `RELEASE_NAMESPACE` value to the namespace where Epinio was
installed into (by default it's `epinio`) and add it to a file of your choice:
```

nunix commented on docs/howtos/create_custom_appcharts.md:
````diff
@@ -0,0 +1,150 @@
+  helmChart: https://mydomain.org/epinio-application-0.1.15-fluentd.tgz
+```
+
+into a file of your choice and change `RELEASE_NAMESPACE` to be the namespace Epinio was
+installed into. The default is `epinio`.
````
```suggestion
```

nunix commented on docs/howtos/create_custom_appcharts.md:
````diff
@@ -0,0 +1,150 @@
+$ epinio app chart list
+```
+
+and/or
````
```suggestion
You can also see the details of the chart in Epinio:
```

nunix commented on docs/howtos/create_custom_appcharts.md:
````diff
@@ -0,0 +1,150 @@
+$ epinio app chart show fluent
+```
+
+If this fails check that the `RELEASE_NAMESPACE` was properly set in the CR.
````
```suggestion
## Troubleshooting
If the new Helm chart is not correctly registered in Epinio, check that the `RELEASE_NAMESPACE` was properly set in the CRD.
```

andreas-kupries replied:
`CRD` is the custom resource definition. `CR` is the custom resource and what we have here. Maybe better to expand the shorthand, or name it differently.

nunix reviewed (commented):
Reviewed `create_custom_appcharts` only. Will have a second review for the `using_custom_appcharts`.

andreas-kupries commented on docs/howtos/create_custom_appcharts.md:
```diff
@@ -0,0 +1,150 @@
+
+Note that explanations about setting up and using `fluentd` are out of scope here.
+
+Use your editor of choice to edit the file `templates/deployment.yaml`.
```
In the clone step one of the commands has the user setting `helm-charts/chart/application` as their working directory.
Thus the shorter relative path here is ok.

We could possibly rewrite the entire how-to to stay either outside the clone, or on the toplevel directory, and then use larger paths across the board. On the good side of that change the user would not have to track state in the form of the working directory.

andreas-kupries commented on docs/howtos/create_custom_appcharts.md:
```diff
@@ -0,0 +1,150 @@
+Note how the templating uses the `.Values.epinio.appName` field to insert the application
+name into the annotation.
+
+The file `values.yaml` in the chart explains the full set of values Epinio sets when
+deploying an application through the chart.
```
See previous response regarding working directory in the executed commands.

andreas-kupries commented:
My plan is now to accept all these, and then re-tweak a few places I noted.

andreas-kupries committed:
Apply suggestions from code review

Co-authored-by: Nuno do Carmo <nuno.carmo@suse.com>

andreas-kupries committed:
Tweaks post adressing comments

- Slight reformatting, refilled to 90 columns.
- Dropped contractions ('s, 'll, ...).
- Dropped changes to working directory as commands are executed.
- Changed paths to be based on the unchanged working directory the repo is cloned into.

andreas-kupries committed:
update cli reference for app charts, and other changes

andreas-kupries committed:
tweak docs to hide the unwanted operator commands from devs

andreas-kupries committed:
feat: added new howtos and cli refs to the sidebar.

andreas-kupries committed:
fix: title headers in cli reference docs

nunix commented on docs/howtos/using_custom_appcharts.md:
````diff
@@ -0,0 +1,109 @@
+$ epinio app chart default
+```
+
+Note that the configured default is also made visible when listing the available charts.
+In the example output below the chart `foo` is marked as the currently active default.
````
```suggestion
You can also check which chart is set as `default` when listing the available charts.
As an example, the output below shows the chart `FOO` as the current default:
```

nunix commented on docs/howtos/using_custom_appcharts.md:
````diff
@@ -0,0 +1,109 @@
+```
+| DEFAULT |   NAME   |        DESCRIPTION         |
+|---------|----------|----------------------------|
+| *       | foo      | Foofy deployment           |
````
```suggestion
| *       | FOO      | Foofy deployment           |
```

nunix commented on docs/howtos/using_custom_appcharts.md:
```diff
@@ -0,0 +1,109 @@
+
+# Unsetting the default Helm chart
+
+To restore the use of the system default (`standard`) invoke the command
```
```suggestion
You can set the system chart `standard` back as the default, by running the following command:
```

nunix commented on docs/howtos/using_custom_appcharts.md:
````diff
@@ -0,0 +1,109 @@
+$ epinio app chart default ''
+```
+
+Note the __empty string__ used in place of the chart name. This is the signal to unset the
+current setting.
````
```suggestion
:::note
The __empty string__ used in place of the chart name is mandatory. If you do not add it,
the command will only list the charts available.
:::
```

nunix reviewed (commented):
Reviewed the `using_custom_appcharts` doc.
Rephrased to active voice and added admonitions.

andreas-kupries commented on docs/howtos/using_custom_appcharts.md:
```diff
@@ -0,0 +1,109 @@
+The argument of the option is the name of custom Helm chart to use, as shown when listing
+the available charts.
+
+__Beware__ that changing the chart to use is __not possible__ if the application has an
+active workload.
```
Thanks, was not aware of this kind of syntax.

andreas-kupries commented on docs/howtos/using_custom_appcharts.md:
````diff
@@ -0,0 +1,109 @@
+```
+| DEFAULT |   NAME   |        DESCRIPTION         |
+|---------|----------|----------------------------|
+|         | standard | Epinio standard deployment |
````
No. `standard` is the __system default__ and is automatically used when the user neither sets a local default, nor uses `--app-chart`. As such it is not set as the local default by default, and is not highlighted.

andreas-kupries commented:
Thanks for the review. I accept all the changes and will then fix a few typos I saw in them.

andreas-kupries committed:
Apply suggestions from code review

Co-authored-by: Nuno do Carmo <nuno.carmo@suse.com>

andreas-kupries committed:
fix: small nits in the previous

andreas-kupries commented:
Pushed my fixes to your fixes now.
